# NGS_Course
Course on analysis of NGS data (RNA-Seq. and ATAC-Seq.) using [Galaxy](https://usegalaxy.eu/).

A good command on basic R is good, but not necessary. The course can be followed with zero prior knowledge of R or any bioinformatics tool.

## Topics Covered
1. File types in NGS (fastq, sam / bam, genome index and gtf / gff3)
2. Mapping Fastq files
3. RNA-Seq. analysis
4. Designing and saving workflow in Galaxy 
5. Controlling for covariates in RNA-Seq. analysis
6. ATAC-Seq. pipeline
7. Interrogating ATAC-Seq. data for peaks, tf binding sites, enriched motifs (possibility with Galaxy not tested. May need R) 

As multiple steps in RNA-Seq. and ATAC-Seq. are time consuming, the entire course will not occur live. For time consuming steps, we will use **'Flipped Classroom'** principles. The sessions with all the steps will be recorded in advance. You would need to go through the session and execute the steps by yourself. Afterwards, we will have live sessions / Zoom meetings to discuss issues and go into details.

