In this session, we will perform detailed analysis of RNA-Seq. count data and provide an introduction to ATAC-Seq. analysis. 

RNA-Seq. Analysis
==================================

Step 1
---------
Generate count files by following [the exercise for Session 2 from Step 1 - 5](https://github.com/sumeetpalsingh/NGS_Course/blob/master/Exercises/Session2_Exercise.md). 

Step 2
--------
Download count data from an experiment conducted in zebrafish. The count data was generated by using the full raw file used in Session 2 and Step 1 above. We will use this file for in-depth analysis of RNA-Seq. data [DGE and GO analysis](https://github.com/sumeetpalsingh/NGS_Course/raw/master/Data/RNASeq_counts.csv): https://github.com/sumeetpalsingh/NGS_Course/raw/master/Data/RNASeq_counts.csv.

Step 3
--------
Have a good internet connection! We will be using [iDEP](http://bioinformatics.sdstate.edu/idep92/) website for RNA-Seq. analysis.


ATAC-Seq. Analysis
================================

Step 4
---------
In this session, we will map ATAC-Seq. raw data to the genome and identify peaks. For this, [download this file and unzip it.](https://github.com/sumeetpalsingh/NGS_Course/raw/master/Data/atac_raw_files.zip): https://github.com/sumeetpalsingh/NGS_Course/raw/master/Data/atac_raw_files.zip

The zip files contains two paired-end fastq files generated in zebrafish. It contains ATAC-Seq. data from two tissues. We will try to identify differentially expressed genes between the two tissues.

With this, you are ready for Session 3!
